docs/arcadia-users-group/20221017-conda/lesson.md

```diff
@@ -0,0 +1,369 @@
+To get access to the conda software, you need to install it.
+The most popular way to install conda is via miniconda or miniforge.
+**[miniforge](https://github.com/conda-forge/miniforge)** is the community (conda-forge) driven minimalistic conda installer. 
+By default, subsequent package installations come from conda-forge channel.
+**[miniconda](https://docs.conda.io/en/latest/miniconda.html)** is the Anaconda (company) driven minimalistic conda installer. 
+By default, subsequent package installations come from anaconda channels (default or otherwise).
+The default channel order can be changed for either conda installation type.
+miniforge started when miniconda didn't support the linux aarch64 system architectures and was quickly adopted by many conda user.
+It's stuck around even though both miniconda and miniforge support most system architectures.
+
+One of the problems conda addresses is resolving dependency conflicts betweens many pieces of software installed in the same environment.
+This is an [NP-complete problem](https://www.anaconda.com/blog/understanding-and-improving-condas-performance) meaning it gets slower as more software is added to an environment and it's a hard problem to solve.
+[mamba](https://mamba.readthedocs.io/en/latest/index.html) is a drop-in replacement for conda that offers higher speed and more reliable environment solutions.
+However, the best way to install mamba at the moment is via conda.
+Mamba is worth the confusion it has caused -- it decreases install times by orders of magnitude thus saving time.
+
+## Extra content
+
+### Conda and workflow managers
+
+Conda is integrated into workflow managers like snakemake and nextflow.
+Using environment files or other directives that point toward the conda packages that are needed for a specific rule or process, the workflow manager will use conda to build the necessary environments and will automatically activate and deactivate them for each rule or process as it runs the workflow.
+In this case, it's best practice to have only the tools needed for a specific rule or process in a given environment -- often, this will mean only including one tool per an environment.
+
+For snakemake, conda environments are encoded in YAML files and fed to a given rule using the `conda:` directive and the path to the YAML file.
+For nextflow, conda environments are actually the least-recommended way to execute a pipeline.
+While you can give a process a conda package name and its build hash, it's better to use the biocontainer build for that conda package on docker.
+This overcomes [issues with reproducibility that stem from differences in operating systems](https://www.nature.com/articles/nbt.3820).
+
+### Building a conda package from an R library on CRAN
+
+Once you become reliant on conda to manage all of your packages, it can be disappointing when your favoRite new pacakge isn't available on the conda-forge channel.
```

Comment:
```suggestion
Once you become reliant on conda to manage all of your packages, it can be disappointing when your favorite new pacakge isn't available on the conda-forge channel.
```